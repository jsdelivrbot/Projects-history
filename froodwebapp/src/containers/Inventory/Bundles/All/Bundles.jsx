import React from 'react';
import { ConnectedMainContainer } from 'containers';
import { connect } from 'react-redux';
import { BUNDLES_REQUESTS } from 'redux-base/actions';
import { getMainContainerState } from 'utils';
import {
  renderItem,
  getRowClickRedirectLink
} from './bundlesHelpers';

const mapStateToProps = state => getMainContainerState(state, 'bundles');

const mapDispatchToProps = BUNDLES_REQUESTS;

export const Bundles = props => (
  <ConnectedMainContainer
    // autocomplete
    autocompletePlaceholder="Search by sku code"
    renderAutocompleteItem={ renderItem }

    // top buttons
    newButtonVisible
    newButtonText="New Bundle"
    newButtonLink="/inventory/item-bundles/new"

    // table
    actionColumnVisible={ false }
    getRowClickRedirectLink={ getRowClickRedirectLink }

    { ...props }
  />
);

export default connect(mapStateToProps, mapDispatchToProps)(Bundles);
