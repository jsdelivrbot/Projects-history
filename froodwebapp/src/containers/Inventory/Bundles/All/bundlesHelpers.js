import { renderAutocompleteItem } from 'utils';

export const getStatisticsColumns = () => ([
  {
    value: 300,
    description: 'Item Bundles'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Active'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Recurring'
  }, {
    value: 15,
    description: 'Since Yesterday',
    isDynamic: true
  }]
);

export const renderItem = renderAutocompleteItem(['id', 'name', 'skuCode'], 'Item Bundle');

export const getRowClickRedirectLink = row => (
  `/inventory/item-bundles/${row.id}`
);
