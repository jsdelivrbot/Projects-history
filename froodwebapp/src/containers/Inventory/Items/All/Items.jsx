import React from 'react';
import { ConnectedMainContainer } from 'containers';
import { connect } from 'react-redux';
import { ITEMS_REQUESTS } from 'redux-base/actions';
import { getMainContainerState } from 'utils';
import {
  renderItem,
  getRowClickRedirectLink
} from './itemsHelpers';

const mapStateToProps = state => getMainContainerState(state, 'items');

const mapDispatchToProps = ITEMS_REQUESTS;

export const Items = props => (
  <ConnectedMainContainer
    // autocomplete
    autocompletePlaceholder="Search by sku code"
    renderAutocompleteItem={ renderItem }

    // top buttons
    newButtonVisible
    newButtonText="New Inventory"
    newButtonLink="/inventory/all-items/new"

    // table
    actionColumnVisible={ false }
    getRowClickRedirectLink={ getRowClickRedirectLink }

    { ...props }
  />
);

export default connect(mapStateToProps, mapDispatchToProps)(Items);
