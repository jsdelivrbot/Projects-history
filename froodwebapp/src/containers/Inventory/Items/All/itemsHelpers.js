import { renderAutocompleteItem } from 'utils';

export const getStatisticsColumns = () => ([
  {
    value: 300,
    description: 'All Items'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Active'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Recurring'
  }, {
    value: 15,
    description: 'Since Yesterday',
    isDynamic: true
  }]
);

export const renderItem = renderAutocompleteItem(['id', 'name', 'skuCode'], 'Item');

export const getRowClickRedirectLink = row => (
  `/inventory/all-items/${row.id}`
);
