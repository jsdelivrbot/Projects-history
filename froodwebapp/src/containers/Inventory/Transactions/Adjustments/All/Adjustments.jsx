import React from 'react';
import { ConnectedMainContainer } from 'containers';
import { connect } from 'react-redux';
import { ADJUSTMENTS_REQUESTS } from 'redux-base/actions';
import { getMainContainerState } from 'utils';
import {
  renderItem,
  getStatisticsColumns
} from './adjustmentsHelpers';

const mapStateToProps = state => getMainContainerState(state, 'adjustments');

const mapDispatchToProps = ADJUSTMENTS_REQUESTS;

export const Adjustments = props => (
  <ConnectedMainContainer
    // autocomplete
    autocompletePlaceholder="Search by adjustment name"
    renderAutocompleteItem={ renderItem }

    // top buttons
    newButtonVisible
    newButtonText="New Adjustment"
    newButtonLink="/inventory/adjustments/new"

    // table
    actionColumnVisible={ false }

    // helpers
    getStatisticsColumns={ getStatisticsColumns }
    { ...props }
  />
);

export default connect(mapStateToProps, mapDispatchToProps)(Adjustments);
