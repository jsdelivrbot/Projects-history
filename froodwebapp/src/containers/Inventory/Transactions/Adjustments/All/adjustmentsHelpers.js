import { renderAutocompleteItem } from 'utils';

export const getStatisticsColumns = () => ([
  {
    value: 300,
    description: 'Adjustments'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Active'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Recurring'
  }, {
    value: 15,
    description: 'Since Yesterday',
    isDynamic: true
  }]
);

export const renderItem = renderAutocompleteItem(['bin', 'sku', 'warehouse'], 'Adjustment');
