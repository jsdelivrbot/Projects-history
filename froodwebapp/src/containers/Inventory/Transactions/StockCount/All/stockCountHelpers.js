import { renderAutocompleteItem } from 'utils';

export const getStatisticsColumns = () => ([
  {
    value: 300,
    description: 'Purchases'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Active'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Recurring'
  }, {
    value: 15,
    description: 'Since Yesterday',
    isDynamic: true
  }]
);

export const renderItem = renderAutocompleteItem(['code', 'startDate', 'endDate'], 'Stock Count');

export const getRowClickRedirectLink = row => (
  `/inventory/stock-count/${row.id}/${row.code}`
);
