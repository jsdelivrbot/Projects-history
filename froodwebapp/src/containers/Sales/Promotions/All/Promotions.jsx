import React from 'react';
import { ConnectedMainContainer } from 'containers';
import { connect } from 'react-redux';
import { PROMOTIONS_REQUESTS } from 'redux-base/actions';
import { getMainContainerState } from 'utils';
import {
  renderItem,
  getRowClickRedirectLink
} from './promotionsHelpers';

const mapStateToProps = state => getMainContainerState(state, 'promotions');

const mapDispatchToProps = PROMOTIONS_REQUESTS;

export const Promotions = props => (
  <ConnectedMainContainer
    // autocomplete
    autocompletePlaceholder="Search by promotion code"
    renderAutocompleteItem={ renderItem }

    // top buttons
    newButtonVisible
    newButtonText="New Promotion"
    newButtonLink="/sales/promotions/new"

    // table
    actionColumnVisible={ false }
    getRowClickRedirectLink={ getRowClickRedirectLink }

    { ...props }
  />
);

export default connect(mapStateToProps, mapDispatchToProps)(Promotions);
