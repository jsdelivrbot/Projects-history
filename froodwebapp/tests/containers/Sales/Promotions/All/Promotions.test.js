import { Promotions } from 'containers/Sales/Promotions/All/Promotions';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('Promotions', () => {
  it('renders the Promotions component', () => {
    const component = shallow(<Promotions />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of Promotions', () => {
    const component = shallow(<Promotions />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
