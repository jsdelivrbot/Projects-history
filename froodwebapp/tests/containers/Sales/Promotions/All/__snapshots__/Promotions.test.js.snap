// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promotions capturing Snapshot of Promotions 1`] = `
<Connect(MainContainer)
  actionColumnVisible={false}
  autocompletePlaceholder="Search by promotion code"
  getRowClickRedirectLink={[Function]}
  newButtonLink="/sales/promotions/new"
  newButtonText="New Promotion"
  newButtonVisible={true}
  renderAutocompleteItem={[Function]}
/>
`;
